MineSweeper.py

```diff
@@ -0,0 +1,254 @@
+    box_index_y = (mouse_click[1] - 10) // BOX_SIZE
+
+    try:
+      self._tiles[box_index_x][box_index_y].marked_mine = True
+
+    except IndexError:
+      pass
+
+    self.draw()
+
+
+class Tile():
+
+  def __init__(self, x, y, size):
+    self._x_index = x
+    self._y_index = y
+    self._x = (x*size) + 10
+    self._y = (y*size)  + 10
+    self._size = size
+
+    self.is_clicked = False
+    self.is_mine = False
+    self.marked_mine = False
+    self.value = 0
+
+  # Tile function to draw itself to the window. 
+  def draw(self, panel):
+
+    # If the tile has yet to be clicked, draw a dark gray box on top of a light gray box
+    # If the tile is marked as a mine, draw a cute lil flag on there
+    if(self.is_clicked == False):
+      panel.create_rectangle(self._x, self._y, self._x + self._size, 
+			     self._y + self._size, fill = 'grey')
+
+      panel.create_line(self._x, self._y, self._x + self._size, 
+			self._y + self._size, fill = 'dim grey')
+
+      panel.create_line(self._x, self._y + self._size, self._x + self._size,
+			self._y, fill = 'dim grey')
+
+      panel.create_rectangle(self._x + 2, self._y + 2, self._x + self._size - 2, 
+			     self._y + self._size - 2, fill = 'dim grey', outline = 'dim grey')
+
+      if(self.marked_mine):
+        panel.create_rectangle(self._x + 5, self._y + 5, self._x + self._size - 5, 
+			       self._y + self._size/2, fill = 'black') 
+
+        panel.create_line(self._x + 5, self._y + 5, self._x + 5, 
+			  self._y + self._size - 2)
+
+    else:
+      if(self.is_mine):
+        panel.create_rectangle(self._x, self._y, self._x + self._size, 
+			       self._y + self._size, fill = 'grey')
+        panel.create_line(self._x, self._y, self._x + self._size, 
+			  self._y + self._size, fill = 'red')
+        panel.create_line(self._x, self._y + self._size, self._x + self._size,
+			  self._y, fill = 'red')
+        panel.create_line(self._x, self._y + self._size/2, self._x + self._size, 
+			  self._y + self._size/2, fill = 'red')
+        panel.create_line(self._x + self._size/2, self._y, self._x + self._size/2, 
+			  self._y + self._size, fill = 'red')
+      
+      else:
+        panel.create_rectangle(self._x, self._y, self._x + self._size, 
+			       self._y + self._size, fill = 'grey')
+
+        if(self.value != 0):
+          panel.create_text(self._x + self._size/2, self._y + self._size/2,
+			         text = str(self.value))
+
+  # Tile function to reveal a tile. 
+  def show_self(self):
+    self.is_clicked = True
+    self.marked_mine = False
+
+  # gathers all real neighbors and returns a list of their indices
+  def get_neighbors(self):
+
+    neighbors = []
+
+    for i in range(-1, 2):
+      for j in range(-1, 2):
+
+        if(0 <= (self._x_index + i) < (BOARD_WIDTH/BOX_SIZE - 1)):
+          if(0 <= (self._y_index + j) < (BOARD_WIDTH/BOX_SIZE -1)):
+            #if(i != 0 and j != 0):
+            neighbors.append([self._x_index + i, self._y_index + j])
```

Comment:
Walk through this loop with a piece of paper and a pencil and write down each step of it, by hand, and you'll see what the issue is.